metrics/collector_test.go

```diff
@@ -125,6 +125,83 @@ func testMachineStatus(t *testing.T) {
 	}
 }
 
+func MachineStatusWhenMachineDeleted(t *testing.T) {
+	testCase := machineStatusTestCase{
```

Comment:
You do not need to use machineStatusTestCase.
We have already checked expectedMetrics with another test.